packages/dtslint/src/index.ts

```diff
@@ -160,38 +174,40 @@ async function runTests(
   let warnings: string[] | undefined;
   let errors: string[] | undefined;
 
-  if (!skipNpmChecks && !expectOnly) {
+  if (npmChecks) {
     ({ implementationPackage, warnings, errors } = await checkNpmVersionAndGetMatchingImplementationPackage(
       packageJson,
       packageDirectoryNameWithVersion,
     ));
   }
 
-  const minVersion = maxVersion(packageJson.minimumTypeScriptVersion, TypeScriptVersion.lowest);
-  if (onlyTestTsNext || tsLocal) {
-    const tsVersion = tsLocal ? "local" : TypeScriptVersion.latest;
-    await testTypesVersion(dirPath, tsVersion, tsVersion, expectOnly, tsLocal, /*isLatest*/ true);
-  } else {
-    // For example, typesVersions of [3.2, 3.5, 3.6] will have
-    // associated ts3.2, ts3.5, ts3.6 directories, for
-    // <=3.2, <=3.5, <=3.6 respectively; the root level is for 3.7 and above.
-    // so this code needs to generate ranges [lowest-3.2, 3.3-3.5, 3.6-3.6, 3.7-latest]
-    const lows = [TypeScriptVersion.lowest, ...typesVersions.map(next)];
-    const his = [...typesVersions, TypeScriptVersion.latest];
-    assert.strictEqual(lows.length, his.length);
-    for (let i = 0; i < lows.length; i++) {
-      const low = maxVersion(minVersion, lows[i]);
-      const hi = his[i];
-      assert(
-        parseFloat(hi) >= parseFloat(low),
-        `'"minimumTypeScriptVersion": "${minVersion}"' in package.json skips ts${hi} folder.`,
-      );
-      const isLatest = hi === TypeScriptVersion.latest;
-      const versionPath = isLatest ? dirPath : joinPaths(dirPath, `ts${hi}`);
-      if (lows.length > 1) {
-        console.log("testing from", low, "to", hi, "in", versionPath);
+  if (npmChecks !== "only") {
```

Comment:
Maybe this is a nit, but if this were written as an early return, I think it may be clearer / less diff-y?

```ts
if (npmChecks === "only") {
   return;
}
```

Reply:
There’s stuff after this block, though

Reply:
Ah, crap. The GitHub diff ended perfectly right after the block and I really thought the function had ended 😄 

![image](https://github.com/microsoft/DefinitelyTyped-tools/assets/5341706/f38ba259-1498-468b-adfb-79e939fef4d2)